src/main/java/lib/motors/TalonFXSim.java

```diff
@@ -0,0 +1,110 @@
+package lib.motors;
+
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.controls.*;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import lib.math.differential.Derivative;
+import lib.units.Units;
+
+public class TalonFXSim {
+
+    private final DCMotorSim motorSim;
+    private final Derivative acceleration = new Derivative();
+    private PIDController controller = null;
+    private ProfiledPIDController profiledController = null;
+    private double lastTimestampSeconds = 0;
+
+    private double voltageRequest;
+
+    public TalonFXSim(int numMotors, double gearing, double jKgMetersSquared) {
+        DCMotor motor = DCMotor.getFalcon500(numMotors);
+
+        motorSim = new DCMotorSim(motor, gearing, jKgMetersSquared);
+    }
+
+    public void update(double timestampSeconds) {
+        motorSim.update(timestampSeconds - lastTimestampSeconds);
+        acceleration.update(getVelocity(), timestampSeconds);
+
+        lastTimestampSeconds = timestampSeconds;
+    }
+
+    public TalonFXSim setController(PIDController controller) {
+        this.controller = controller;
+        return this;
+    }
+
+    public TalonFXSim setProfiledController(ProfiledPIDController profiledController) {
+        this.profiledController = profiledController;
+        return this;
+    }
+
+    public StatusCode setControl(DutyCycleOut request) {
+        return this.setControl(new VoltageOut(request.Output * 12));
+    }
+
+    public StatusCode setControl(VoltageOut request) {
+        voltageRequest = request.Output;
+        motorSim.setInputVoltage(voltageRequest);
+        return StatusCode.OK;
```

Comment:
If you always return `OK` you can just make this method `void`.

Reply:
The whole point was to make sure that this class would be exactly like using a real TalonFX. This means that if we want to use the error codes to check something (maybe logging of configurations), the simulation could support this as well. I'm not saying that this is necessary with the current code, but it's consistent with the TalonFX class, and that could be useful.

Reply:
My point is that always returning `OK`, regardless of whether the operation succeeded, doesn't provide useful information.
I also think it's unnecessary to emulate 100% of the behavior of real hardware. All inputs and outputs must behave the same for the replay to work, but I'm not sure that simulating configurations status codes is that important. Some of `TalonFXSim`'s methods do return status codes so that you can use them.

Reply:
I changed it

Reply:
Did you opt then to not check status codes at all?